lib/types/measures/MeasureContent.ts

```diff
@@ -24,85 +19,76 @@ export interface AssetMeasurement {
   /**
    * Property containing the actual measurement.
    *
-   * This should be specialized by child interfaces
+   * This should be specialized by child interfaces.
    */
   values: JSONObject;
 
   /**
-   * Micro Timestamp of the measure
+   * Micro Timestamp of the measurement time.
    */
-  measuredAt?: number;
-}
+  measuredAt: number;
 
-/**
- * Represents a measurement of a value to post from a decoder
- *
- * This interface should be extended and the `values` property specialized
- * to declare new measurement type.
- */
-export interface Measurement {
   /**
-   * A device may have different measures for the same type (e.g. measure temperature 2 times)
-   * Should be set when you link the device to the asset
+   * Name given by the decoder to the measure.
    */
-  name?: string;
-
-  /**
-   * Property containing the actual measurement.
-   *
-   * This should be specialized by child interfaces
-   */
-  values: JSONObject;
+  deviceMeasureName: string;
+}
 
+export interface AssetMeasurement extends Measurement {
   /**
-   * Micro Timestamp of the measurement time
+   * Name given by the `deviceLink` of the linked asset.
    */
-  measuredAt: number;
+  assetMeasureName: string;
 }
 
 /**
  * Represent the full content of a measure document.
  */
-export interface MeasureContent extends KDocumentContent, Measurement {
-  /**
-   * Type of the measure. (e.g. "temperature")
-   * The type name is also the name of the sub-property to look at
-   * in the "values" object to get the measure main value.
-   */
-  type: string;
-
+export interface MeasureContent extends KDocumentContent, AssetMeasurement {
   /**
-   * Measurement self-description
+   * Measurement self-description.
    */
   unit: MeasureUnit;
 
   /**
-   * Origin of the measure
+   * Define the origin of the measure.
    */
   origin: {
     /**
-     * ID of the device (document _id)
+     * From what the measure has been pushed. Can be :
+     * - 'asset'
+     * - 'device'
      */
-    id?: string;
+    type: OriginType;
 
     /**
-     * E.g. "device"
+     * Payload uuid that was used to create this measure.
      */
-    type: string;
+    payloadUuid?: string;
```

Comment:
Since a measure can be computed from many payloads, this should remains an array

Reply:
Was it already used as an array? Or is it for future use? And is there really a case where a Measure could be composed of multiple payloads instead of having tinier measure definition ?

Reply:
Done